### PR TITLE
pkg/release: add conditionSet

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4355.
Towards https://github.com/giantswarm/giantswarm/issues/4337.

### CODE REVIEW

File: pkg/release/condition_set.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"context"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (kopiczko) [inline pkg/release/condition_set.go]: I don't like that we have to pull k8s.io deps to release package. But there is a helm client anyway, so I'm not very sad.

File: pkg/release/release.go [hunk 2]
```diff
@@ -32,6 +35,8 @@ type Release struct {
 	logger     micrologger.Logger
 
 	namespace string
+
+	condition *conditionSet
```
Comment (kopiczko) [inline pkg/release/release.go]: Ready to use in InstallOperator.

File: pkg/release/condition_set.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+type conditionFn func() error
+
+type conditionSetConfig struct {
+	ExtClient *apiextensionsclient.Clientset
+	Logger    micrologger.Logger
+}
+
+type conditionSet struct {
```
Comment (kopiczko) [inline pkg/release/condition_set.go]: `conditionSet` to not colide with `condition`.

File: pkg/release/condition_set.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		logger:    config.Logger,
+	}
+
+	return c, nil
+}
+
+func (c *conditionSet) CRD(ctx context.Context, name string) conditionFn {
+	return func() error {
+		o := func() error {
+			_, err := c.extClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(name, metav1.GetOptions{})
```
Comment (xh3b4sd) [inline pkg/release/condition_set.go]: Do we have something similar in `operatorkit` already? Can we combine or refactor something there somehow? Just a quick brainwave.

File: pkg/release/release.go [hunk 5]
```diff
@@ -22,6 +24,7 @@ type Config struct {
 	ApprClient *apprclient.Client
 	HelmClient *helmclient.Client
 	Logger     micrologger.Logger
+	RestConfig *rest.Config
```
Comment (xh3b4sd) [inline pkg/release/release.go]: IMO this is a setting and not a dependency.

File: pkg/release/condition_set.go (hunk 4 above)
Comment (kopiczko) [reply pkg/release/condition_set.go]: I don't think so. If you can point out exactly what it is we can maybe use it.
But I'm not so keen to pull operatorkit to testing libraries.

File: pkg/release/release.go (hunk 5 above)
Comment (kopiczko) [reply pkg/release/release.go]: Done.

File: pkg/release/release.go [hunk 6]
```diff
@@ -23,7 +25,8 @@ type Config struct {
 	HelmClient *helmclient.Client
 	Logger     micrologger.Logger
 
-	Namespace string
+	Namespace  string
+	RestConfig *rest.Config
```
Comment (MarcelMue) [inline pkg/release/release.go]: I am not sure if we want to pass the rest config in here, or pass the individual clients. How do we want to handle this? Now both `host` and this gets a rest config and creates its own clients.

Comment (MarcelMue) [reply pkg/release/release.go]: I would prefer just creating the clients in one place and passing them around tbh. Uncertain.

Comment (kopiczko) [reply pkg/release/release.go]: I was just worried about additional changes when we add more clients.
But maybe we should add all at once?

Comment (MarcelMue) [reply pkg/release/release.go]: You are right, let's pass the restconfig for now and do more changes later

Comment (kopiczko) [reply pkg/release/release.go]: All the clients added.

Comment (MarcelMue) [reply pkg/release/release.go]: Ack, should be fine. We'll see.

**Review by MarcelMue — APPROVED** [review]
LGTM

**Pull request merged**